### ISSUE TITLE
bin/test is not using the Rails test reporter

### ISSUE THREAD

**Issue by kirs** [rails/rails#29899]
Since https://github.com/rails/rails/pull/29572, the output of `bin/test` and `bundle exec ruby -Itest ...` is slightly different output. `bin/test` doesn't load the Rails minitest plugin, which leads to minitest' default reporter used. Note that I'm talking about the rails/rails test suite.

I'm not 100% sure, but I think it happens because `bin/test` is calling `Rails::TestUnit::Runner.run(ARGV)` as another one is using `Minitest.run(ARGV)`

### Expected behavior

`bundle exec ruby -Itest test/cases/my_test.rb` and `bin/test test/cases/my_test.rb` should have the same output.

### Actual behavior

`bundle exec ruby -Itest test/cases/my_test.rb` is using shiny new Rails test reporter and prints rerun command:

<img width="741" alt="screen shot 2017-07-23 at 23 01 29" src="https://user-images.githubusercontent.com/522155/28502433-31ed546a-6ffb-11e7-9090-01a189a41d7d.png">

`bin/test test/cases/my_test.rb` is using minitest built-in reporter:

<img width="995" alt="screen shot 2017-07-23 at 23 01 41" src="https://user-images.githubusercontent.com/522155/28502445-4abf75f4-6ffb-11e7-8c3e-e6e7111d0518.png">

**Comment by kirs** [comment]
@kaspth you were talking about dogfooding the test reporter, but it looks like `bin/test` doesn't invoke it. I'll be happy to fix it if you can give me any lead.

**Comment by kirs** [comment]
More deviation in backtrace when running a failing test in railties:

With `bin/test`:

<img width="878" alt="screen shot 2017-07-24 at 22 19 44" src="https://user-images.githubusercontent.com/522155/28540511-4fd9a852-70be-11e7-8aa7-1b2e69f4fe10.png">

With `bundle exec ruby -Itest ...`:

<img width="844" alt="screen shot 2017-07-24 at 22 19 31" src="https://user-images.githubusercontent.com/522155/28540526-608049ae-70be-11e7-90ee-c21c0ab593de.png">

**Comment by utilum** [comment]
The only distinction I see is in backtrace absolute path (bin) *vs.* relative path... what am I missing?

```
$ rails test test/models/post_test.rb --backtrace
Running via Spring preloader in process 10176
Run options: --backtrace --seed 58218

# Running:

F

Failure:
PostTest#test_the_lie [/home/oz/code/blog/test/models/post_test.rb:9]:
Expected true to not be truthy.


bin/rails test test/models/post_test.rb:8

.

Finished in 0.018579s, 107.6462 runs/s, 107.6462 assertions/s.
2 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

```
$ bundle exec ruby -Itest test/models/post_test.rb --backtrace 
Run options: --backtrace --seed 30290

# Running:

F

Failure:
PostTest#test_the_lie [test/models/post_test.rb:9]:
Expected true to not be truthy.


bin/rails test test/models/post_test.rb:8

.

Finished in 0.016244s, 123.1220 runs/s, 123.1220 assertions/s.
2 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

**Comment by kirs** [comment]
@utilum I'm talking about Rails tests, not the application tests. See ActiveRecord examples in the issue description.

**Comment by pvalena** [comment]
I have found some more deviations caused AFAICT by https://github.com/rails/rails/commit/0d72489b2a08487f71dd4230846c01a5d99ef35f, f.e.:

 - output contains `bin/rails test` instead of `bin/test`
 - line filters do not work

**Comment by y-yagi** [comment]
Fixed by #30318
